instrumentation/spring/spring-webflux/spring-webflux-5.3/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/v5_3/internal/HeaderUtil.java

```diff
@@ -32,6 +39,33 @@ public final class HeaderUtil {
             () ->
                 findGetHeadersMethod(
                     MethodType.methodType(List.class, Object.class))); // before spring web 7.0
+
+    // Spring Web 7+
+    MethodHandle forEach = null;
+    try {
+      forEach =
+          MethodHandles.lookup()
+              .findVirtual(
+                  HttpHeaders.class,
+                  "forEach",
+                  MethodType.methodType(void.class, BiConsumer.class));
+    } catch (Throwable t) {
+      // ignore - will fall back to keySet
+    }
+    FOR_EACH = forEach;
+
+    // Spring Web 6 and earlier
+    MethodHandle keySet = null;
+    if (FOR_EACH == null) {
+      try {
+        keySet =
+            MethodHandles.lookup()
+                .findVirtual(Map.class, "keySet", MethodType.methodType(Set.class));
+      } catch (Throwable t) {
+        // ignore
+      }
+    }
+    KEY_SET = keySet;
```

Comment:
if `getKeys` has branches for different versions then might as well cast headers to Map there and invoke the method directly